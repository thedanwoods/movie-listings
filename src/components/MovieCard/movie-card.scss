.movie-card {
  background-color: #000;
  color: #bbb;
  max-width: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 8px 0;
}

.movie-card__poster-container {
  // Reserve space for image whilst loading
  position: relative;
  padding-bottom: 150%;
  height: 0;
  width: 100%;
  max-width: 500px;
}

.movie-card__poster {
  position: absolute;
  width: 100%;
  overflow: hidden;
}

.movie-card__info {
  padding: 16px;
  text-align: center;
  text-transform: uppercase;
  overflow-wrap: break-word;
}

.movie-card__title {
  font-size: 19px;
  padding: 0 0 8px;
  font-family: 'Montserrat', sans-serif;
  font-weight: 400;
}

.movie-card__genres {
  font-family: 'Montserrat', sans-serif;
  font-weight: 400;
  font-size: 11px;
}
