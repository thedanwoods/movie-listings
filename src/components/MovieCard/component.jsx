import React from 'react';
import PropTypes from 'prop-types';

import './movie-card.scss';

const MovieCard = ({ movie, genres }) => (
  <div className="movie-card">
    <div className="movie-card__poster-container">
      <img
        className="movie-card__poster"
        src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`}
        alt={`${movie.title} poster`}
      />
    </div>
    <section className="movie-card__info">
      <h1 className="movie-card__title">{movie.title}</h1>
      <p className="movie-card__genres">
        {genres.length && genres.join(', ')}
      </p>
    </section>
  </div>
);

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  genres: PropTypes.array.isRequired,
};

export default MovieCard;
