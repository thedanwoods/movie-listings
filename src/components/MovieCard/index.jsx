import { connect } from 'react-redux';
import MovieCard from './component';

export const mapStateToProps = (state, ownProps) => ({
  genres: state.genres
    .filter(genre => ownProps.movie.genre_ids.includes(genre.id))
    .map(genre => genre.name),
});

export default connect(mapStateToProps)(MovieCard);
