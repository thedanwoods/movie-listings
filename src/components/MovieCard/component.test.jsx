import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './component';

const defaultProps = {
  movie: { poster_path: '/abc', title: 'Mock title' },
  genres: [{ id: 1, name: 'Action' }],
};

describe('<MovieCard /> component', () => {
  it('renders without errors', () => {
    const movieCard = shallow(<MovieCard {...defaultProps} />);

    expect(() => movieCard).not.toThrow();
    expect(movieCard.debug()).toMatchSnapshot();
  });
});
