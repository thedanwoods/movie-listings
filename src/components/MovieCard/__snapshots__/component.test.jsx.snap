// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> component renders without errors 1`] = `
"<div className=\\"movie-card\\">
  <div className=\\"movie-card__poster-container\\">
    <img className=\\"movie-card__poster\\" src=\\"https://image.tmdb.org/t/p/w500/abc\\" alt=\\"Mock title poster\\" />
  </div>
  <section className=\\"movie-card__info\\">
    <h1 className=\\"movie-card__title\\">
      Mock title
    </h1>
    <p className=\\"movie-card__genres\\">
      [object Object]
    </p>
  </section>
</div>"
`;
